addons/survey/tests/test_survey_invite.py

```diff
@@ -194,3 +194,26 @@ def test_survey_invite_token_internal(self):
             set(answers.mapped('email')),
             set([self.user_emp.email]))
         self.assertEqual(answers.mapped('partner_id'), self.user_emp.partner_id)
+
+    def test_survey_invite_token_by_email_nosignup(self):
+        Answer = self.env['survey.user_input']
+
+        self.env['res.partner'].create({
+            'name': 'Test 1',
+            'email': 'test@example.com',
```

Comment:
Just to make test more interesting we could define two different emails, with normalized email being the same, like
```
Customer1 <test@example.com>
"Customer Duplicate" <TEST@example.COM>
```